src/App.tsx

```diff
@@ -1,14 +1,74 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Option } from './types/Option';
+
+interface FilterOptions {
+  select: string;
+  query: string;
+}
+
+function filterBy(todos: Todo[], { select, query }: FilterOptions): Todo[] {
+  let filtered = todos;
```

Comment:
You should add a copy of the array instead of defining a reference to the array
```suggestion
  let filteredTodos = [...todos];
```